File: app/src/main/java/es/santirivera/surveilfall/di/modules/RepositoryModule.java
package es.santirivera.surveilfall.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import es.santirivera.surveilfall.data.net.NetworkManager;
import es.santirivera.surveilfall.data.net.WServices;
import es.santirivera.surveilfall.data.repository.AppRepository;
import es.santirivera.surveilfall.data.repository.AppRepositoryImpl;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    public AppRepository provideAppRepository(
            Context context,
            WServices wServices,
            NetworkManager networkManager) {

        return new AppRepositoryImpl(context,
                wServices,
                networkManager);
    }

}

File: app/src/main/java/es/santirivera/surveilfall/di/modules/RepositoryModule.kt
package es.santirivera.surveilfall.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides
import es.santirivera.surveilfall.data.net.GithubWebServices
import es.santirivera.surveilfall.data.net.ScryfallWebServices
import es.santirivera.surveilfall.data.repository.AppRepository
import es.santirivera.surveilfall.data.repository.AppRepositoryImpl
import es.santirivera.surveilfall.data.repository.DBRepository
import es.santirivera.surveilfall.data.repository.DatabaseRepository
import javax.inject.Singleton

@Module
class RepositoryModule {

    @Provides
    @Singleton
    fun provideAppRepository(
            scryfall: ScryfallWebServices,
            github: GithubWebServices): AppRepository {
        return AppRepositoryImpl(scryfall, github)
    }

    @Provides
    @Singleton
    fun provideDBRepository(context: Context): DBRepository {
        return DatabaseRepository(context)
    }

}
